src/Collection.ts

```diff
@@ -356,15 +346,31 @@ export default abstract class Collection<T extends Model> extends Base {
    * use the options to disable adding, changing
    * or removing.
    */
-  fetch({ data, ...otherOptions }: SetOptions = {}): Request {
-    const { abort, promise } = apiClient().get(this.url(), data, otherOptions)
+  fetch({ data, ...otherOptions }: SetOptions = {}): Promise<any> {
+    const { promise } = apiClient().get(this.url(), data, otherOptions)
 
-    promise
+    return promise
```

Comment:
we can now use `await + try/catch` instead of `then + catch`

Reply:
In a another PR